modules/mapper-extras/src/test/java/org/elasticsearch/index/mapper/extras/MatchOnlyTextFieldMapperTests.java

```diff
@@ -65,6 +91,13 @@ protected void registerParameters(ParameterChecker checker) throws IOException {
         );
     }
 
+    private XContentBuilder testMapping() throws IOException {
+        if (randomBoolean()) {
+            return syntheticSourceMapping(b -> b.startObject("field").field("type", "match_only_text").endObject());
+        }
+        return fieldMapping(b -> b.field("type", "match_only_text"));
+    }
+
```

Comment:
I think I'd be happier with coverage here if we explicitly run both tests for the synthetic and non-synthetic case, this feels at the moment like we're only testing 50% of the functionality on each run and I think that will come back to bite us.

Reply:
It has a lovely caching behavior that I kind of miss here. Like, if two queries need this then it'll cache.